### ISSUE TITLE
Button to load more new items does not consistently get un-hidden for keyboard or screen reader users.

### ISSUE THREAD

**Issue by MarcoZehe** [nolanlawson/pinafore#59]
I am having real trouble with the button that should come up whenever there are newer items to load, for example newer notifications, items in the Home timeline etc. The button to load them often remains hidden when it should become visible. When tabbing through the page, and there are definitely new items, I sometimes just focus the heading itself and press Space, and the new items then get loaded. But if new items are there, the button should appear and be actionable even without having to tab to it explicitly. I don't even know how mouse users do it, or know that something new is there they should reload...

**Comment by nolanlawson** [comment]
Thanks for reporting! It looks like I added [`role="alert"` to this element](https://github.com/nolanlawson/pinafore/blob/b3e859f89d08907bf4e4bd932deec457b6b388f4/routes/_components/timeline/MoreHeader.html#L1), and I'm also toggling [`aria-hidden` on its parent element](https://github.com/nolanlawson/pinafore/blob/b3e859f89d08907bf4e4bd932deec457b6b388f4/routes/_components/virtualList/VirtualListHeader.html#L2), so maybe it's the combination of those two things that is causing problems. Perhaps it shouldn't even have `role="alert"`?

I can test this out with VoiceOver and a keyboard on a Mac to see what's going on. Is that a good way to reproduce your issue?

**Comment by MarcoZehe** [comment]
Yes. Note that, if you use proper CSS style hiding, like visibility:hidden; or display:none;, aria-hidden is not even needed. aria-hidden cuts the whole tree from the node it is set on, including all its children, so if the children contain focusable elements, this is actually quite dangerous. aria-hidden should not be used unless absolutely necessary. :)

role=„alert“ is actually not bad in itself, since this would cause VoiceOver and other screen readers to read once Pinafore detects new items and updates the button. But the aria-hidden is probably killing something there. aria-hidden is a real beast. ;)

**Comment by nolanlawson** [comment]
Thanks for the tip. I occasionally use `aria-hidden` for performance reasons when "hiding" elements using `opacity: 0; pointer-events: none`. But it's probably unnecessary here. I'll keep digging, thanks again for helping find these issues!

**Comment by MarcoZehe** [comment]
Yeah I know it’s tempting. ;) But especially when you have focusable items in the children that get hidden, if they really become focusable before the accessibility tree has been updated after the removal of aria-hidden, things can get messy. Again, using VoiceOver and the keyboard on the Mac is a good way to test this. Again, there are times when this does work, but more often than not, it does not.

**Comment by nolanlawson** [comment]
I have a PR to remove the `role=dialog` (https://github.com/nolanlawson/pinafore/pull/68), which I believe may be exacerbating the problem. It also doesn't appear to be very helpful on active timelines (it's a bit too chatty 😊).

I just realized that the reason I was doing the `aria-hidden`/`opacity`/`pointer-events` trick is because I'm doing an animation to fade the button in and out. It could be that that's also causing problems, since there's a slight delay between when `aria-hidden` gets toggled and when `pointer-events` gets toggled, which may make it hard to focus.

By any chance can you tell me which screen reader you're using? I'm not sure I can reproduce the issue in VoiceOver.

I'm going to keep digging on this, but if all else fails I can orchestrate the animations such that we don't need `aria-hidden`.

**Comment by MarcoZehe** [comment]
I use VoiceOver, which is the only screen reader on the Mac. But I have also seen inconsistencies with NVDA and Firefox on Windows with this.
